backend/mem/file.go

```diff
@@ -0,0 +1,444 @@
+package mem
+
+import (
+	"errors"
+	"io"
+	"path"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//File implements vfs.File interface for the in-memory implementation of FileSystem.
+type File struct {
+	exists       bool
+	lastModified time.Time
+	isOpen       bool
+	contents     []byte       //the file contents
+	name         string       //the base name of the file
+	cursor       int          //the index that the buffer (contents) is at
+	location     vfs.Location //the location that the file exists on
+}
+
+/*		******* Error Functions *******		*/
+func doesNotExist() error {
+	return errors.New("This file does not exist!")
+}
+
+func copyFailNil() error {
+	return errors.New("The target file passed in was nil")
+}
+
+func seekError() error {
+	return errors.New("Seek could not complete the desired call")
+}
+
+/*		******************		*/
+
+//Close imitates io.Closer by resetting the cursor and setting a boolean
+func (f *File) Close() error {
+
+	if f == nil {
+		return errors.New("Cannot close a nil file")
+	}
+
+	f.isOpen = false
+	f.cursor = 0
+	return nil
+}
+
+//Read implements the io.Reader interface.  Returns number of bytes read and potential errors
+func (f *File) Read(p []byte) (n int, err error) {
+
+	//if file exists:
+	offset := f.cursor
+	if f.isOpen == false {
+		f.isOpen = true
+	}
+	existence, err := f.Exists()
+	if !existence {
+		return 0, err
+	}
+	readBufLen := len(p)
+	if readBufLen == 0 { //readBufLen of byte slice is zero, just return 0 and nil
+		return 0, nil
+	}
+	fileContentLength := len(f.contents)
```

Comment:
Contrary to how I probably generally come off in these discussions I'm actually fine with using small, even single-letter variables (the Go Way) as long as the variable's use is limited to a 5 to maybe... 10-20 line function/block. But if you are going to spell out a variable, spell it out. Why shortcut 2-4 characters when not doing so guarantees legibility?

My main reasons for this is: 1. there is no ambiguity or personal opinion for what words/letters to abbreviate, and what will be obvious to everyone, and 2. it is, granted, probably obvious what readBufLen means when we look at it, but if I'm adding to or refactoring this later, and I'm doing something with your 'readBufLen' variable, I have to remember how you abbreviated it... is it 'readBufferLength'? 'readBuffLen'? 'readBufLength'? etc... better to spell it out entirely, or if the scope is limited enough, abbreviate it absolutely: 'l' In this case, you're working with multiple 'lengths' so just 'l' might be a bad example...

Reply:
Sorry, this may not be the best place to have the ongoing variable naming conversation... mostly I agree with Judd's comment, as far as this specific request goes...